libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -777,3 +891,72 @@ proc heartbeat*(g: GossipSub) {.async: (raises: [CancelledError]).} =
     for trigger in g.heartbeatEvents:
       trace "firing heartbeat event", instance = cast[int](g)
       trigger.fire()
+
+when defined(libp2p_gossipsub_1_4):
+  proc preambleExpirationHeartbeat*(
```

Comment:
Maybe we could extract the `heartbeat` call to when this function is used and have the function just be a single run?